crates/streamer/src/ascii_streamer.rs

```diff
@@ -0,0 +1,68 @@
+use std::future::Future;
+
+use tracing::instrument;
+
+use crate::Signal;
+
+use tokio::io::{AsyncBufReadExt, BufReader};
+
+pub trait AsciiStreamSocket {
+    fn send(&mut self, value: String) -> impl Future<Output = anyhow::Result<()>> + Send;
+    fn close(self);
+}
+
+#[instrument(skip_all)]
+pub async fn ascii_stream(
+    mut websocket: impl AsciiStreamSocket,
+    input_stream: impl tokio::io::AsyncRead + tokio::io::AsyncSeek + Unpin, // A file usually
```

Comment:
question: Do you really need `AsyncSeek`?

Reply:
updated